discovery_engine/test/feed_manager_test.dart

```diff
@@ -196,4 +199,67 @@ Future<void> main() async {
       // doc3 is excluded since it is inactive
     });
   });
+
+  group('Excluded sources', () {
+    setUp(() async {
+      final source = Uri(host: 'www.bbc.com');
+      await excludedSourcesRepo.save({source});
+    });
+
+    tearDown(() async {
+      await excludedBox.clear();
+    });
+
+    test('addExcludedSource', () async {
+      final excludedSoures = {
+        Uri(host: 'www.bbc.com'),
+        Uri(host: 'www.nytimes.com')
+      };
+
+      final source1 =
+          Uri.parse('https://www.bbc.com/politics/87654321-example-article');
+      final source2 =
+          Uri.parse('https://www.nytimes.com/live/2022/03/08/world/example');
+
+      final response1 = await mgr.addExcludedSource(source1);
+      final response2 = await mgr.addExcludedSource(source2);
+
+      expect(response1, isA<ClientEventSucceeded>());
+      expect(response2, isA<ClientEventSucceeded>());
+      expect(excludedBox.values.first, equals(excludedSoures));
+    });
+
+    test('removeExcludedSource', () async {
+      final excludedSoures = {
+        Uri(host: 'www.bbc.com'),
+        Uri(host: 'www.nytimes.com')
+      };
+      await excludedSourcesRepo.save(excludedSoures);
+
+      final source =
+          Uri.parse('https://www.bbc.com/politics/87654321-example-article');
+      final response = await mgr.removeExcludedSource(source);
+
+      expect(response, isA<ClientEventSucceeded>());
+      expect(excludedBox.values.first, equals({Uri(host: 'www.nytimes.com')}));
+    });
+
+    test('getExcludedSourcesList', () async {
+      final excludedSoures = {
+        Uri(host: 'theguardian.com'),
+        Uri(host: 'bbc.co.uk'),
+        Uri(host: 'wsj.com'),
```

Comment:
Will it be a problem that `wsj.com` and `www.wsj.com` lead to two entries?

Reply:
Will it match subdomains?
Eg www.bbc.co.uk is excluded
The article is then sport.bbc.co.uk ?

Reply:
It depends on NewCatcher. This will be an exact match on the source value. If they return articles with `www.example.com` and `example.com` then they will treated as different sources

Reply:
the `www` subdomain is a "special" case, so we could consider to threat it as equal to apex domain, for example if we find that NewsCatcher sometimes returns articles from the same source with `www` and other times without it. For now I would leave it separate cases.

Reply:
Why special? It has the same dignity of `sport`. We cannot control this data is all up to the provider at this point.

Reply:
it's special in a sence that `www` and non-`www` domains are most of the time, for most websites, connected with each other through 301 redirects, which is not the case for other subdomains like `sport`. But like I said before, we can rely on the data from newscatcher, and this could be easily changed later on.

Reply:
@acrrd I don't see any actionable items here for the PR, is it ok to resolve this conversation?

Reply:
NC indeed always presents urls before any redirects, it actually makes sense indeed to just always drop www.

Eg, they crawl https://bbc.co.uk, and then find a relative link there which is /sport

Then, it might be that the crawled one actually becomes https://www.bbc.co.uk/sport after resolving it to an absolute link.

If you always drop www, then this is no issue

Reply:
...and then https://www.bbc.co.uk/sport redirects maybe to sport.bbc.co.uk, but that then doesn't matter

Reply:
I think it might depend how this set of sources is used. If we use send it with requests to the API so they are doing filtering on their side, or if we use it to filter out documents we got back. In the first case we probably shouldn't change anything and rely on data received from newscatcher, in the second case we might want to always drop the `www`.

Reply:
The only valid values are the one in `source_domain` we cannot do anything with them.